authentication/authentication_request.go

```diff
@@ -0,0 +1,201 @@
+		default:
+			return nil, err
+		}
+	}
+
+	return response, nil
+}
+
+// Request combines NewRequest and Do, while also handling decoding of response payload.
+func (a *Authentication) Request(ctx context.Context, method, uri string, payload interface{}, resp interface{}) error {
+	request, err := a.NewRequest(ctx, method, uri, &payload)
+	if err != nil {
+		return fmt.Errorf("failed to create a new request: %w", err)
+	}
+
+	response, err := a.Do(request)
+	if err != nil {
+		return fmt.Errorf("failed to send the request: %w", err)
+	}
+	defer response.Body.Close()
+
+	// If the response contains a client or a server error then return the error.
+	if response.StatusCode >= http.StatusBadRequest {
+		return newError(response)
+	}
+
+	responseBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read the response body: %w", err)
+	}
+
+	if len(responseBody) > 0 && string(responseBody) != "{}" {
+		if err = json.Unmarshal(responseBody, &resp); err != nil {
+			return fmt.Errorf("failed to unmarshal response payload: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// NewFormRequest returns a new HTTP request. If the payload is not nil it will be
+// encoded as JSON.
+func (a *Authentication) NewFormRequest(
+	ctx context.Context,
+	method,
+	uri string,
+	payload url.Values,
+	opts ...RequestOption,
+) (*http.Request, error) {
+	request, err := http.NewRequestWithContext(ctx, method, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	for _, option := range opts {
+		option.apply(request, payload)
+	}
+
+	body := strings.NewReader(payload.Encode())
+
+	request.Body = io.NopCloser(body)
+	request.ContentLength = int64(body.Len())
+
+	return request, nil
+}
+
+// FormRequest combines FormRequest and Do, while also handling decoding of response payload.
+func (a *Authentication) FormRequest(ctx context.Context, method, uri string, payload url.Values, resp interface{}, opts ...RequestOption) error {
+	request, err := a.NewFormRequest(ctx, method, uri, payload, opts...)
+	if err != nil {
+		return fmt.Errorf("failed to create a new request: %w", err)
+	}
+
+	response, err := a.Do(request)
+	if err != nil {
+		return fmt.Errorf("failed to send the request: %w", err)
+	}
+	defer response.Body.Close()
+
+	// If the response contains a client or a server error then return the error.
+	if response.StatusCode >= http.StatusBadRequest {
+		return newError(response)
+	}
+
+	responseBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read the response body: %w", err)
+	}
+
+	if len(responseBody) > 0 && string(responseBody) != "{}" {
+		if err = json.Unmarshal(responseBody, &resp); err != nil {
+			return fmt.Errorf("failed to unmarshal response payload: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// RequestOption configures a call (typically to retrieve a resource) to Auth0 with
+// query parameters.
```

Comment:
```suggestion
// RequestOption configures a call to Auth0 with query parameters.
```
That part is only applicable to the Management API.